src/sentry/static/sentry/app/components/events/errors.tsx

```diff
@@ -0,0 +1,242 @@
+  state: State = {
+    isOpen: false,
+  };
+
+  shouldComponentUpdate(nextProps: Props, nextState: State) {
+    if (this.state.isOpen !== nextState.isOpen) {
+      return true;
+    }
+    return this.props.event.id !== nextProps.event.id;
+  }
+
+  toggle = () => {
+    this.setState({isOpen: !this.state.isOpen});
+  };
+
+  uniqueErrors = errors => uniqWith(errors, isEqual);
+
+  onReprocessEvent = async () => {
+    const {orgId, project, event} = this.props;
+    const endpoint = `/projects/${orgId}/${project.slug}/events/${event.id}/reprocessing/`;
+
+    addLoadingMessage(t('Reprocessing event\u2026'));
+
+    try {
+      await this.props.api.requestPromise(endpoint, {
+        method: 'PUT',
+      });
+    } catch (e) {
+      clearIndicators();
+      addErrorMessage(
+        t('Failed to start reprocessing. The event is likely too far in the past.')
+      );
+      return;
+    }
+
+    clearIndicators();
+    browserHistory.push(
+      `/organizations/${orgId}/issues/?query=tags[original_event_id]:${event.id}`
+    );
+  };
+
+  onReprocessGroup = async () => {
+    const {orgId, issueId} = this.props;
+    const endpoint = `/organizations/${orgId}/issues/${issueId}/reprocessing/`;
+
+    addLoadingMessage(t('Reprocessing issue\u2026'));
+
+    try {
+      await this.props.api.requestPromise(endpoint, {
+        method: 'PUT',
+      });
+    } catch (e) {
+      clearIndicators();
+      addErrorMessage(
+        t('Failed to start reprocessing. The event is likely too far in the past.')
+      );
+      return;
+    }
+
+    clearIndicators();
+    browserHistory.push(
+      `/organizations/${orgId}/issues/?query=tags[original_group_id]:${issueId}`
+    );
+  };
+
+  onReprocessStart = () => {
+    openModal(this.renderReprocessModal);
+  };
+
+  renderReprocessModal = ({Body, closeModal}) => (
+    <React.Fragment>
+      <Body>
+        <p>
+          {t(
+            'You can choose to re-process events to see if ' +
+              'your errors have been resolved. Keep the following limitations in mind:'
+          )}
+        </p>
+
+        <ul>
+          <li>
+            {t(
+              'Sentry will duplicate events in your project (for now) and not delete the old versions.'
+            )}
+          </li>
+          <li>
+            {t(
+              'Reprocessing one or multiple events counts against your quota, but bypasses rate limits.'
+            )}
+          </li>
+          <li>
+            {t(
+              'If an event is reprocessed but did not change, we will not create the new version and not bill you for it (for now).'
+            )}
+          </li>
+          <li>
+            {t(
+              'If you have provided missing symbols please wait at least 1 hour before attempting to re-process. This is a limitation we will try to get rid of.'
+            )}
+          </li>
+        </ul>
+      </Body>
+      <div className="modal-footer">
+        <Button onClick={this.onReprocessGroup}>
```

Comment:
You can wrap this in a <ButtonBar gap={1}> so that you don't have to do inline marginLefts.

Reply:
will do